seisbench/models/dkpn.py

```diff
@@ -0,0 +1,784 @@
+import json
+from typing import Any
+
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from packaging import version
+
+import seisbench.util as sbu
+from seisbench.models.base import WaveformModel, _cache_migration_v0_v3
+
+# ---------  For PreProc class
+from obspy.core import Trace
+import copy
+from obspy.signal.filter import bandpass
+from scipy.signal import lfilter
+
+
+class DKPN(WaveformModel):
+    """
+    .. document_args:: seisbench.models PhaseNet
+    """
+
+    _annotate_args = WaveformModel._annotate_args.copy()
+    _annotate_args["*_threshold"] = ("Detection threshold for the provided phase", 0.3)
+    _annotate_args["blinding"] = (
+        "Number of prediction samples to discard on each side of each window prediction",
+        (0, 0),
+    )
+    _annotate_args["overlap"] = (_annotate_args["overlap"][0], 1500)
+
+    _weight_warnings = [
```

Comment:
This is unnecessary here. It warns about outdated weights versions but for DKNP we do not have these.